File: app/src/main/java/com/BlizzardArmory/BlizzardArmory.java
package com.BlizzardArmory;

import android.app.Application;
import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

public class BlizzardArmory extends Application {

    private static BlizzardArmory instance;

    @Override
    public void onCreate() {
        super.onCreate();

        if (instance == null) {
            instance = this;
        }
    }

    public static BlizzardArmory getInstance() {
        return instance;
    }

    public static boolean hasNetwork() {
        return instance.isNetworkConnected();
    }

    private boolean isNetworkConnected() {
        ConnectivityManager cm =
                (ConnectivityManager) getSystemService(Context.CONNECTIVITY_SERVICE);

        NetworkInfo activeNetwork = cm.getActiveNetworkInfo();
        return activeNetwork != null &&
                activeNetwork.isConnectedOrConnecting();
    }
}

File: app/src/main/java/com/BlizzardArmory/BlizzardArmory.kt
package com.BlizzardArmory

import android.app.Application
import android.content.Context
import android.util.Log
import com.akexorcist.localizationactivity.core.LocalizationApplicationDelegate
import java.util.*

/**
 * The type Blizzard armory.
 */
class BlizzardArmory : Application() {

    private val localizationDelegate = LocalizationApplicationDelegate()

    override fun attachBaseContext(base: Context) {
        localizationDelegate.setDefaultLanguage(base, Locale.ENGLISH)
        super.attachBaseContext(localizationDelegate.attachBaseContext(base))
    }

    override fun onCreate() {
        super.onCreate()
        if (instance == null) {
            instance = this
        }
        Thread.setDefaultUncaughtExceptionHandler { _, throwable ->
            Log.e("Crash Prevented", throwable.message!!)
        }
    }

    companion object {
        /**
         * Gets instance.
         *
         * @return the instance
         */
        var instance: BlizzardArmory? = null
            private set
    }
}